le-taxi-api-tests/src/taxis/invalidTaxi.apiTest.ts

```diff
@@ -35,6 +35,33 @@ export async function invalidTaxiTests(): Promise<void> {
     );
   });
 
+  it('Should return bad request when receiving an invalid object', async () => {
+    await shouldThrow(
+      () => postTaxi(`{ "data": [null] }` as any),
+      err => {
+        assert.strictEqual(err.status, StatusCodes.BAD_REQUEST);
+        assert.strictEqual(err.response.body.error.message, `Cannot instantiate invalid object`);
+      }
+    );
+  });
+
+  it('Should return bad request when trying to override the `constructor` property', async () => {
+    const dtoCreate = copyTaxiTemplate();
+    const data = { ...dtoCreate.data[0], constructor: 'InvalidConstructor' };
+    dtoCreate.data[0] = data;
+
+    await shouldThrow(
+      () => postTaxi({ data: [{ constructor: 'InvalidConstructor' }] } as any),
+      err => {
+        assert.strictEqual(err.status, StatusCodes.BAD_REQUEST);
+        assert.strictEqual(
+          err.response.body.error.message,
+          `Cannot instantiate an object with an overriden 'constructor' property`
```

Comment:
Most case would be that they intend to update vehicle by this endpoint instead of /api/vehicles/:id . Should not we tell them to update vehicle via /api/vehicles/:id ?

Reply:
If they are actually sending a vehicleDto on the wrong endpoint I think that this is explicit enough. Trying to identify the type of DTO they are sending on each endpoint feels like falling a down a rabbit hole. It opens the door to many other checks we might want to do - and where would we stop?